sessionctx/variable/session.go

```diff
@@ -1171,6 +1173,15 @@ func (s *SessionVars) GetStatusFlag(flag uint16) bool {
 	return s.Status&flag > 0
 }
 
+// SetInTxn sets whether the session is in transaction.
+// It also updates the IsExplicit flag in TxnCtx if val is true.
+func (s *SessionVars) SetInTxn(val bool) {
+	s.SetStatusFlag(mysql.ServerStatusInTrans, val)
+	if val {
+		s.TxnCtx.IsExplicit = true
```

Comment:
Is it reset when startinging a new transaction?

Reply:
I think it is always reset by `PrepareTxnCtx` or `NewTxn`. I am not sure if there is anything missing.

However, at least the unit test in this PR passes, which I think covers the important cases.